### PR TITLE
[NCL-7960] enable promotion tracking

### PR DESCRIPTION
**Pull request by michalovjan**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -312,6 +312,15 @@
       </releases>
     </repository>
 
+    <repository>
```
Comment (pkocandr) [inline pom.xml:315]: Why do you add central in here?

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java [hunk 2]
```diff
@@ -702,15 +706,19 @@ private void promoteDownloads(PromotionPaths promotionPaths, Runnable heartBeatS
...
+            Runnable heartBeatSender,
+            String promotionTrackingID) throws RepositoryDriverException, PromotionValidationException {
         for (SourceTargetPaths sourceTargetPaths : promotionPaths.getSourceTargetsPaths()) {
             heartBeatSender.run();
             try {
                 PathsPromoteRequest request = new PathsPromoteRequest(
                         sourceTargetPaths.getSource(),
                         sourceTargetPaths.getTarget(),
                         sourceTargetPaths.getPaths());
+                request.setTrackingId(promotionTrackingID);
```
Comment (pkocandr) [inline src/main/java/org/jboss/pnc/repositorydriver/Driver.java:721]: I got a bit confused here. There is a `setPromotionId()` method, but this seems to be added recently, so it was probably just an identification of the promotion, not the ID for promotion tracking.

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java [hunk 3]
```diff
@@ -254,7 +253,7 @@ public void promote(@SpanAttribute(value = "promoteRequest") RepositoryPromoteRe
                 heartBeatSender.run();
                 PromotionPaths downloadsPromotions = trackingReportProcessor
                         .collectDownloadsPromotions(report, genericRepos);
-                promoteDownloads(downloadsPromotions, heartBeatSender, promoteRequest.isTempBuild());
+                promoteDownloads(downloadsPromotions, heartBeatSender, promoteRequest.isTempBuild(), buildContentId);
```
Comment (pkocandr) [inline src/main/java/org/jboss/pnc/repositorydriver/Driver.java:256]: Shouldn't we use here `"build-" + buildContentId` as we do for the uploads/downloads tracking ID?

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java [hunk 4]
```diff
@@ -263,7 +262,8 @@ public void promote(@SpanAttribute(value = "promoteRequest") RepositoryPromoteRe
                                 buildType.getRepoType(),
                                 buildContentId),
                         promoteRequest.isTempBuild(),
-                        heartBeatSender);
+                        heartBeatSender,
+                        buildContentId);
```
Comment (pkocandr) [inline src/main/java/org/jboss/pnc/repositorydriver/Driver.java:266]: DTTO

File: pom.xml (hunk 1 above)
Comment (michalovjan) [reply pom.xml:315]: For 2 reasons:
1. Having nexus-staging in front of central makes every query go to staging which slows down fetching dependencies. This is mainly just annoying.
2. Having nexus-staging in front of central creates a potential of downloading shady artifacts from staging even though they are present in central. (had this happen to me in the past)

It just makes more sense in general, as staging should behave as: Lets try to look in staging if the artifact has not yet been released.

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java (hunk 2 above)
Comment (michalovjan) [reply src/main/java/org/jboss/pnc/repositorydriver/Driver.java:721]: I've particularly looked at Henry's PRs in Indy/Indy-promote-service to find out which IDs I should use.

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java (hunk 3 above)
Comment (michalovjan) [reply src/main/java/org/jboss/pnc/repositorydriver/Driver.java:256]: That's my bad. I've assumed buildContentId is already in form of build-XXXXXXXX.

File: pom.xml (hunk 1 above)
Comment (pkocandr) [reply pom.xml:315]: OK, good points.

File: src/main/java/org/jboss/pnc/repositorydriver/Driver.java (hunk 3 above)
Comment (pkocandr) [reply src/main/java/org/jboss/pnc/repositorydriver/Driver.java:256]: You are right, the format is already like that. Sorry for the confusion.

**Review by pkocandr — APPROVED** [review]
All good.

**Pull request merged**